.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,58 @@
+name: Run tests
+
+on:
+  push:
+  release:
+    types:
+      - published
+
+jobs:
+  php-tests:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        php: [7.2, 7.3, 7.4]
+        laravel: [6.*, 7.*, 8.*]
+        os: [ubuntu-latest]
+        coverage: ['none', 'xdebug']
+
+    name: '[P${{ matrix.php }}] [L${{ matrix.laravel }}] [${{ matrix.coverage }}]'
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v1
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, bcmath, soap, intl, gd, gmp, exif, iconv, imagick
+          coverage: ${{ matrix.coverage }}
+          ini-values: memory_limit=-1
+          tools: composer:v2
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v2"
+        with:
+          path: |
+            ~/.composer/cache
+            vendor
```

Comment:
It's not advised to cache vendor directly, this can create hard to track problems.

In fact, setup-php has the most recommended way, please see https://github.com/shivammathur/setup-php#cache-composer-dependencies

Reply:
Thanks for the clarification